### PR TITLE
Update Raspberry Pi dockerfile toolchain link

### PR DESCRIPTION
**Pull request by JoseSanchez7**
Update old invalid toolchain link to newer, working link.

# Checklist
- [ x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/main/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x ] This pull-request is submitted against the `main` branch. 
  - [ x] I have merged the latest `main` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem
The link for the raspberry pi toolchain was no longer valid.

# Description of the solution
Update the link to the most recent, stable toolchain release.

### CODE REVIEW

**Review by ericwol-msft — APPROVED** [review]
Do you also want to delete the other cross compile doc in this PR?

**Comment by ericwol-msft** [comment]
[Delete SDK_cross_compile_example.md](https://github.com/Azure/azure-iot-sdk-c/pull/2397/commits/0b0f3549b7beab3d733c42fb5f15dc7569785820)

Is there any links this that also need to be deleted? hint... "YES"

File: samples/dockerbuilds/RaspberryPi/Dockerfile [hunk 1]
```diff
@@ -26,11 +26,11 @@ RUN mkdir RPiBuild
 ENV WORK_ROOT=/home/builder/RPiBuild
 WORKDIR ${WORK_ROOT}
 
-RUN wget https://releases.linaro.org/components/toolchain/binaries/7.3-2018.05/arm-linux-gnueabihf/gcc-linaro-7.3.1-2018.05-x86_64_arm-linux-gnueabihf.tar.xz
-RUN tar -xvf gcc-linaro-7.3.1-2018.05-x86_64_arm-linux-gnueabihf.tar.xz
+RUN wget https://releases.linaro.org/components/toolchain/binaries/latest-7/arm-linux-gnueabihf/gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf.tar.xz
```
Comment (danewalton) [inline samples/dockerbuilds/RaspberryPi/Dockerfile:29]: will this link break if 7.5.0 is no longer the latest (the `latest-7` part).

**Review by danewalton — APPROVED** [review]

**Comment by JoseSanchez7** [comment]
> Do you also want to delete the other cross compile doc in this PR?



> [Delete SDK_cross_compile_example.md](https://github.com/Azure/azure-iot-sdk-c/pull/2397/commits/0b0f3549b7beab3d733c42fb5f15dc7569785820)
> 
> Is there any links this that also need to be deleted? hint... "YES"

Do you mean deleting any other links that link to this file being deleted?

File: samples/dockerbuilds/RaspberryPi/Dockerfile [hunk 2]
```diff
@@ -26,11 +26,11 @@ RUN mkdir RPiBuild
 ENV WORK_ROOT=/home/builder/RPiBuild
 WORKDIR ${WORK_ROOT}
 
-RUN wget https://releases.linaro.org/components/toolchain/binaries/latest-7/arm-linux-gnueabihf/gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf.tar.xz
-RUN tar -xvf gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf.tar.xz
+RUN wget https://releases.linaro.org/components/toolchain/binaries/7.3-2018.05/arm-linux-gnueabihf/gcc-linaro-7.3.1-2018.05-x86_64_arm-linux-gnueabihf.tar.xz
```
Comment (ericwol-msft) [inline samples/dockerbuilds/RaspberryPi/Dockerfile:29]: No, why are you pointing to an older version? Just don't use latest-7. Latest points to 7.5.0-2019.12
The path should be https://releases.linaro.org/components/toolchain/binaries/7.5-2019.12/arm-linux-gnueabihf/gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf.tar.xz

**Comment by ericwol-msft** [comment]
are you going to delete the [SDK_cross_compile_example.md](https://github.com/Azure/azure-iot-sdk-c/pull/2397/files#diff-2d63de2ab3284d925528c4606d540c83e96a4d0bd19a77a4d1b58c714ffdc027) links in this PR?

**Review by ericwol-msft — APPROVED** [review]

**Pull request merged**